src/ASM/ElmMats.C

```diff
@@ -65,39 +65,48 @@ const Vector& ElmMats::getRHSVector () const
 }
 
 
-void ElmMats::printMat (std::ostream& os, size_t idx) const
+void ElmMats::printMat (std::ostream& os, size_t idx,
+                        const char* prefix) const
 {
   if (idx >= A.size()) return;
 
-  os <<"\nElement coefficient matrix";
+  os << (prefix ? prefix : "\nElement coefficient matrix");
```

Comment:
prefix need to handle the newline?

Reply:
The assumption is that the user will provide the newline in prefix, if that is wanted (not necessarily always the case).